### ISSUE TITLE
Regent: add support for generating futures from raw C API handles

### ISSUE THREAD

**Issue by elliottslaughter** [StanfordLegion/legion#673]
Proposal: add a call `__future(T, x)` which takes a value `x` of type `legion_future_t` and returns a value of type `T`. When `-ffuture 1` is enabled, this can be optimized to `future(T)`, giving us a path where C API future values can be passed into Regent.

**Comment by elliottslaughter** [comment]
An example of a test program that should work after this change:

```
import "regent"
task f(x : int) return x + 1 end
task main()
  var y : int = 123
  var z = regentlib.c.legion_future_from_untyped_pointer(__runtime(), &y, [terralib.sizeof(int)])
  var w = __future(int, z)
  regentlib.assert(f(w) == 124, "test failed")
end
```

**Comment by elliottslaughter** [comment]
Merged via 548c01f12f192524f9f2428d34ee885669304201. Thanks @rupanshusoi!
